### PR TITLE
Add multibyte support for attributeSource getOptionId method

### PR DESCRIPTION
**Pull request by gomencal**
Add multibyte support for attributeSource getOptionId method

### Description (*)
Currently, if a option value has special characters in UTF-8 (for instance), the use of strcasecmp PHP function causes incorrect matching, because it doesn't support multibyte strings.

### Manual testing scenarios (*)
1. Create an option value with special characters ("Sí" for instance)
2. In a script try to load it as:
`
$value = 'Sí';
$attribute = $this->resource->getAttribute($attributeCode);
$optionId = $attribute->getSource()->getOptionId($value);
`
It must find the option value and return its Id.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=22033) <br/>All committers have signed the CLA.

File: app/code/Magento/Eav/Model/Entity/Attribute/Source/AbstractSource.php [hunk 1]
```diff
@@ -166,4 +166,22 @@ public function toOptionArray()
...
     }
+
+    /**
+     * Multibyte support strcasecmp function version
+     * @param string $str1
+     * @param string $str2
+     * @param null|string $encoding
+     * @return int|\\lt
+     */
+    private function mbStrcasecmp($str1, $str2, $encoding = null)
```
Comment (aleron75) [inline app/code/Magento/Eav/Model/Entity/Attribute/Source/AbstractSource.php]: Hello,
why do you add an optional parameter if it's never used? Or am I missing something?
Thanks!

Comment (aleron75) [reply app/code/Magento/Eav/Model/Entity/Attribute/Source/AbstractSource.php]: Hello @gomenca, do you have any update on this? Thanks!

Comment (gomencal) [reply app/code/Magento/Eav/Model/Entity/Attribute/Source/AbstractSource.php]: Hi @aleron75 , you are right. I've just used this function in another places and used this parameter. I drop it.

**Review by aleron75 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @aleron75, thank you for the review.
[ENGCOM-4735](https://jira.corp.magento.com/browse/ENGCOM-4735) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@gomencal thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by VasylShvorak** [comment]
:heavy_check_mark: QA passed

**Pull request merged**